### PR TITLE
Fix #8975:  TypingError raised when calling any of the np.random distributions with size being an empty tuple

### LINKED ISSUE #8975: Numba RandomState overload fails with empty size
## Reporting a bug

- [x] I have tried using the latest released version of Numba (most recent is
 visible in the change log (https://github.com/numba/numba/blob/main/CHANGE_LOG).
- [x] I have included a self contained code sample to reproduce the problem.
  i.e. it's possible to run as 'python bug.py'.

```python
import numpy as np
import numba

@numba.njit
def uniform():
    return np.random.uniform(0, 1, size=())

uniform()  # numba.core.errors.TypingError: Failed in nopython mode pipeline (step: nopython frontend)
```

### PR DESCRIPTION
**Pull request by NSchiffmacher**
Closes #8975.

Add a check in `np_uniform_impl3` for size=(), based on @kc611's [suggestion](https://github.com/numba/numba/issues/8975#issuecomment-1564048175).

However, calling `np.random.uniform(0, 1, size=None)` and  `np.random.uniform(0, 1, size=())` does not result in the same output. While `np.random.uniform(0, 1, size=None)` returns a single float, `np.random.uniform(0, 1, size=())` returns an array of shape () and size 1. 

I believe that this is the issue @masaaldosey ran into in #8997.

As this is my first-ever PR (and thus contribution) to an open source project, I'm open to any tips/suggestions/remarks 😄

### CODE REVIEW

**Comment by gmarkall** [comment]
Many thanks for the PR! I think this looks like a good first contribution :-)

The changes look appropriate, and it's excellent that tests are included. It looks good to me but I'm not that familiar with this area, so I'll leave it as ready for review now (and ask what @kc611 thinks in the triage meeting today).

**Comment by NSchiffmacher** [comment]
Thanks for the feedback! 

I also figured that the issue seems to be the same with many (all?) distributions from np.random actually... And this patch only fixes np.random.uniform.

Should I try to do something about the other distributions in this PR, or is it fine if this one gets reviewed, etc and I'll try to fix the other ones in another PR?

**Comment by NSchiffmacher** [comment]
Update: I fixed this behavior for all (I hope) distributions, and added a test that checks them all at once, based on the `TestNumPyRandomAPI` test.

File: docs/upcoming_changes/9636.bug_fix.rst [hunk 1]
```diff
@@ -0,0 +1,5 @@
+Fix calls to ``numpy.random`` distributions with ``size=()``
+------------------------------------------------------------
+
+Calling any of the ``numpy.random`` distributions with ``size=()`` used
```
Comment (kc611) [inline docs/upcoming_changes/9636.bug_fix.rst:4]:
```suggestion
Calling any of the ``numpy.random`` distributions with ``size=()`` is
now supported; previously it used to raise a ``TypingError`` while
being supported by NumPy.
```

**Review by kc611 — COMMENTED** [review]
Hi @NSchiffmacher, this PR looks fantastic. Thanks for covering all the distributions in detail. Minor nitpick below. Otherwise LGTM!

**Comment by NSchiffmacher** [comment]
Hey @kc611, I wasn't sure how to phrase the bug fix in the upcoming changes note, thanks for the fix and the feedback!

File: docs/upcoming_changes/9636.bug_fix.rst [hunk 2]
```diff
@@ -0,0 +1,7 @@
+Fix calls to ``numpy.random`` distributions with ``size=()``
+------------------------------------------------------------
+
+Calling any of the ``numpy.random`` distributions with ``size=()`` is
+now supported; previously it used to raise a ``TypingError`` while
+being supported by NumPy.
+to raise a ``TypingError``, while this behavior is supported by NumPy.
```
Comment (kc611) [inline docs/upcoming_changes/9636.bug_fix.rst:7]:
```suggestion

```

**Review by kc611 — APPROVED** [review]

**Comment by NSchiffmacher** [comment]
Just pushed a minor update to the release note, one of the lines got duplicated @kc611 😅

**Comment by kc611** [comment]
Yep,that's alright. :+1: 

I'll go over the contents once more just in case; but otherwise this will be marked as `Ready to Merge` once the tests suite finishes running.

**Comment by NSchiffmacher** [comment]
Perfect, thank you!

**Pull request merged**